itest/assets_test.go

```diff
@@ -1168,11 +1176,20 @@ func createAssetInvoice(t *testing.T, dstRfqPeer, dst *HarnessNode,
 
 	t.Logf("Got quote for %v asset units per BTC", rate)
 
-	assetUnits := rfqmath.NewBigIntFixedPoint(assetAmount, 0)
-	numMSats := rfqmath.UnitsToMilliSatoshi(assetUnits, *rate)
-	mSatPerUnit := float64(decodedInvoice.NumMsat) / float64(assetAmount)
+	var mSatPerUnit float64
+
+	if cfg.msats > 0 {
+		require.EqualValues(t, decodedInvoice.NumMsat, cfg.msats)
```

Comment:
I don't think this will always be an exact amount? Since we're doing a full msat -> asset -> msat round trip. So I think we should allow for at least a full asset unit (expressed in msat) worth of difference.
And to validate that, we should probably choose an msat amount in the test that is not evenly divisible by the rate.